services/src/test/scala/org/apache/gearpump/services/SubmitApplicationRequestServiceSpec.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.services
+
+import org.apache.gearpump.cluster.MasterToClient.SubmitApplicationResultValue
+import org.apache.gearpump.partitioner.Partitioner
+import org.apache.gearpump.streaming.ProcessorDescription
+import org.apache.gearpump.streaming.appmaster.{StreamingAppMasterDataDetail, SubmitApplicationRequest}
+import org.apache.gearpump.util.{Graph, LogUtil}
+import org.scalatest.prop.PropertyChecks
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import org.slf4j.Logger
+import spray.http.{ContentTypes, HttpEntity}
+import spray.testkit.ScalatestRouteTest
+import upickle._
+
+import scala.concurrent.duration._
+
+class SubmitApplicationRequestServiceSpec extends FlatSpec with PropertyChecks with ScalatestRouteTest with SubmitApplicationRequestService with AppMasterService with Matchers {
+  private val LOG: Logger = LogUtil.getLogger(getClass)
+  def actorRefFactory = system
+  def master = TestCluster.master
+
+  val dag = Graph.empty[Int,String]
+  List(0,2,9,5,8,1,4,11,6,10,3,7).foreach(dag.addVertex)
```

Comment:
Ok, yes i can mock the object in this way. For a functional suite we should perhaps submit different DAG's and verify the response DAG's are in agreement. For this UT I will just verify the JSON has been deserialized as expected into something that client.submitApplication would expect.